workers/data_refinery_workers/processors/tximport.R

```diff
@@ -0,0 +1,63 @@
+# This script is based on the one written by J. Taroni:
+# https://github.com/jaclyn-taroni/ref-txome/blob/master/4-athaliana_tximport.R
+#
+# Two required arguments:
+# (1) --exp_dir    <experiement_dir> (directory where "quant.sf" files are found)
+# (2) --gene2txmap <path_of_gene2txmap_file>
+#
+# Two Output files:
+# (1) RDS file (saved as <experiment_dir>/txi_out.RDS)
+# (2) TPM file (saved as <experiment_dir>/gene_lengthScaledTPM.tsv)
+
+# Handle input arguments:
+library("optparse")
+option_list <- list(
+  make_option("--exp_dir", type = "character"),
+  make_option("--gene2txmap", type = "character")
+)
+
+opt_parser <- OptionParser(option_list = option_list)
+opt <- parse_args(opt_parser)
+
+# directory of the experiment data
+exp.directory <- opt$exp_dir
+
+# Each sample's data is in a sub-dir of experiment directory.
+samples <- list.dirs(exp.directory, full.names = FALSE, recursive = FALSE)
+
+# Get a list of "quant.sf" files in experiment directory.
+# The path of each "quant.sf" are in this format:
+# <exp_dir>/<sample>/processed/quant.sf
+sf.files <- list.files(exp.directory, pattern="^quant.sf$", full.names = TRUE, recursive = TRUE)
+
+# Name each "sf.files" entry with the corresponding sample's name.
+names(sf.files) <- samples
+
+# Generate the gene to transcript mapping based on the input file
+# specified by "--gene2txmap" option:
+gene2tx.file <- opt$gene2txmap
+gene2tx.df <- readr::read_tsv(gene2tx.file, col_names = c("gene_id", "tx_name", "DROP"))
```

Comment:
The way we're doing this now, there shouldn't be trailing tabs. So I think this should do it:
```R
gene2tx_df <- readr::read_tsv(gene2tx_file, col_names = c("gene_id", "tx_name))
```
And you won't need line 40.

Reply:
Sure.